### DISCUSSION TITLE
Eslint "no-undeclared-env-vars"  turbo.json > tasks > build > env  "... is not listed as a dependency in turbo.json"

### DISCUSSION THREAD

**Discussion by zhengzhu-1m** [vercel/turborepo#8536] [Help] [answered]
### Describe the Bug  
https://turbo.build/repo/docs/reference/configuration#env
![image](https://github.com/vercel/turbo/assets/130019078/5e30d00e-13d3-4950-aae2-cc5a5ef77c8d)
```
 error  NEXT_PUBLIC_XXXXXXX is not listed as a dependency in turbo.json       turbo/no-undeclared-env-vars
```

### Expected Behavior 
I want the rule turbo/no-undeclared-env-vars check if variable exist in tasks.build.env

**Comment by anthonyshew** [comment, marked as answer]
That is indeed how the plugin works, but Framework Inference is not currently compatible. I'm working on this here: https://github.com/vercel/turbo/pull/8505
